src/extensions/spikeamplitudes/SpikeAmplitudesView/SpikeAmplitudesPanel.ts

```diff
@@ -0,0 +1,155 @@
+                }
+            }
+        }
+        else if (this._calculationError) {
+            painter.drawText({xmin: timeRange.min, xmax: timeRange.max, ymin: 0, ymax: 1}, {Horizontal: 'AlignCenter', Vertical: 'AlignCenter'}, font, pen, brush, this._calculationError.message)
+        }
+        else {
+            painter.drawText({xmin: timeRange.min, xmax: timeRange.max, ymin: 0, ymax: 1}, {Horizontal: 'AlignCenter', Vertical: 'AlignCenter'}, font, pen, brush, 'calculating')
+            this._scheduleCalculation()
+        }
+    }
+    label() {
+        return this.args.unitId + ''
+    }
+    amplitudeRange() {
+        if (this._amplitudes) {
+            return {min: Math.min(...this._amplitudes), max: Math.max(...this._amplitudes)}
+        }
+        else return null
+    }
+    autoYRange() {
+        if ((this._panelGroup) && (this._panelGroup.length > 0)) {
+            const amplitudeRanges = this._panelGroup.map(p => p.amplitudeRange()).filter(r => (r !== null))
+            if (amplitudeRanges.length > 0) {
+                const ampMin = Math.min(...amplitudeRanges.map(r => r?.min || 0))
+                const ampMax = Math.max(...amplitudeRanges.map(r => r?.max || 0))
+                return {min: ampMin, max: ampMax}
+            }
+        }
+        return this.amplitudeRange()
+    }
+    setYRange(yrange: {min: number, max: number}) {
+        this._yrange = yrange
+    }
+    setPanelGroup(panels: SpikeAmplitudesPanel[]) {
+        this._panelGroup = panels
+    }
+    register(onUpdate: () => void) {
+        this._updateHandler = onUpdate
+    }
+    _scheduleCalculation() {
+        if (this._calculationScheduled) return
+        this._calculationScheduled = true
+        setTimeout(async () => {
+            try {
+                await this._doCalculation()
+            }
+            catch(err) {
+                this._calculationError = err
+            }
+            this._updateHandler && this._updateHandler()
+            this._calculationScheduled = false
+        }, 10)
+    }
+    async _doCalculation() {
+        const result = await this.args.hither.createHitherJob(
+            'createjob_fetch_spike_amplitudes',
+            {
+                recording_object: this.args.recording.recordingObject,
+                sorting_object: this.args.sorting.sortingObject,
+                unit_id: this.args.unitId
+            },
+            {
+                newHitherJobMethod: true,
+                useClientCache: true,
+                required_files: {
+                    sortingObject: this.args.sorting.sortingObject,
+                    recordingObject: this.args.recording.recordingObject
+                }
+            }
+        ).wait()
+        this._timepoints = result.timepoints as number[]
+        this._amplitudes = result.amplitudes as number[]
+        this._maxAmplitude = Math.max(...this._amplitudes)
+    }
+}
+
+class CombinedPanel {
```

Comment:
The CombinedPanel is a panel that combines a collection of panels into one (so they overlap).